vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java

```diff
@@ -126,9 +126,10 @@ public CheckboxGroup() {
             }
             return (CheckboxGroupListDataView) dataView;
         }
-        throw new IllegalStateException(
-                "Required ListDataProvider, but got " + getDataProvider()
-                        .getClass().getSuperclass().getSimpleName());
+        throw new IllegalStateException(String.format(
+                "Required ListDataProvider, but got '%s'. Use 'getDataView()' "
+                        + "to get a generic DataView instance.",
```

Comment:
I don't know if I'm missing something @caalador @mshabarov , but I'd like to repeat why I think it is problematic to store the data view instance in the component:
1) Developer sets `setDataSource("foo", "bar")`
-> `dataView` is set to `CheckboxGroupListDataView` because of `return getListDataView();`
2) Developer calls `setDataSource(myBackendDataProvider);` 
-> this returns `getDataView()` which will have return type of `CheckBoxGroupDataView` but actual type is `CheckboxGroupListDataView` because `dataView` is stored, and it won't work for the data provider type used
3) If the developer now calls `getListDataView()` they get an exception message that tells to use `getDataView()`, which will not work either since the `dataView` is still cached to be the wrong type.

This is why I think the data view storing is problematic and error prone, but am I really alone with this ?
(This also applies for other components where you want to store the data view type)

Reply:
There was a reason I'm quite certain, but removed it and only kept the exception message for better directing the user on what steps to take.

Reply:
Thank you for explanations @pleku , now I see the problem either. After introduction of `getDataView` it is problematic to switch between the in-memory, lazy and generic implementations. Moreover, the data view concept is a some kind of lightweight facade for data handling, so data views aren't a heavy objects and would be created easily.